tests/common/test_classes.py

```diff
@@ -70,14 +79,11 @@ def test_schedule_runs_function(self, inmem_client, flag):
         flag.wait(timeout=5)
         assert flag.is_set()
 
-        # NOTE: Testing threaded code is not fun.
-        # wait until we've received a new message in our inmem client
-        #  messaging backend.
         try:
-            inmem_client._messaging.wait(mailbox=inmem_client.scheduler_mailbox_name, timeout=2)
-            inmem_client._messaging._wait_until_messages_processed()
-        except queue.Empty:
-            # Maybe it's been processed already... just continue anyway then.
+            inmem_client._storage.wait_for_job_update(job_id, timeout=2)
+        except Exception:
```

Comment:
Pattern alert! We risk having a test that work because they just eat up any exception type :)